tracing-subscriber/src/fmt/span.rs

```diff
@@ -223,6 +229,15 @@ fn id_to_idx(id: &Id) -> usize {
     id.into_u64() as usize - 1
 }
 
+#[inline]
+fn find_last_no_duplicate(stack: &[ContextId]) -> Option<&Id> {
+    stack
+        .iter()
+        .rev()
+        .find(|context_id| !context_id.duplicate)
+        .map(|context_id| &context_id.id)
```

Comment:
This iteration probably doesn't have all that significant a perf impact — in the best case, we will only call `Iterator::next` a single time; even if duplicate enters are on the stack, we will probably only have to iterate a couple of times. 👍

Reply:
that's what I though :)